tests/unit/test-debug-info-test.js

```diff
@@ -0,0 +1,208 @@
+import { module, test } from 'qunit';
+import { run } from '@ember/runloop';
+import DebugInfo, { getDebugInfo } from '@ember/test-helpers/-internal/debug-info';
+import MockStableError, { overrideError, resetError } from './utils/mock-stable-error';
+import { MockConsole, getRandomBoolean, getMockDebugInfo } from './utils/test-isolation-helpers';
+
+module('TestDebugInfo', function() {
+  test('summary returns minimal information when debugInfo is not present', function(assert) {
+    assert.expect(1);
+
+    let hasPendingTimers = getRandomBoolean();
+    let hasPendingWaiters = getRandomBoolean();
+    let hasRunLoop = getRandomBoolean();
+    let hasPendingRequests = Boolean(Math.floor(Math.random(10)) > 0);
+    let testDebugInfo = new DebugInfo(
+      hasPendingTimers,
+      hasRunLoop,
+      hasPendingWaiters,
+      hasPendingRequests
+    );
+
+    assert.deepEqual(testDebugInfo.summary, {
+      hasPendingRequests,
+      hasPendingTimers,
+      hasPendingWaiters,
+      hasRunLoop,
+    });
+  });
+
+  test('summary returns full information when debugInfo is present', function(assert) {
+    assert.expect(1);
+
+    run.backburner.DEBUG = false;
+
+    let testDebugInfo = new DebugInfo(
+      false,
+      false,
+      false,
+      false,
+      getMockDebugInfo(false, 2, [{ name: 'one', count: 1 }, { name: 'two', count: 1 }])
+    );
+
+    assert.deepEqual(testDebugInfo.summary, {
+      autorunStackTrace: undefined,
+      hasPendingRequests: false,
+      hasPendingTimers: false,
+      hasPendingWaiters: false,
+      hasRunLoop: false,
+      pendingScheduledQueueItemCount: 2,
+      pendingScheduledQueueItemStackTraces: ['STACK', 'STACK'],
+      pendingTimersCount: 2,
+      pendingTimersStackTraces: ['STACK', 'STACK'],
+    });
+  });
+
+  if (getDebugInfo()) {
```

Comment:
This should use that extracted helper that I mentioned backburnerSupportsDebugInfo() instead, then in its beforeEach set the DEBUG flag. Otherwise this test may or may not be actually ran depending on the ambient state of the debug macro.

Reply:
Yerp.